### PR TITLE
Complete backward doc

### PR DESCRIPTION
**Pull request by Canpio**
update backward documents

### CODE REVIEW

**Comment by Canpio** [comment]
Some thing wrong in this pr. close it.

File: paddle/framework/backward.md [hunk 1]
```diff
@@ -1,23 +1,53 @@
...
+## Backward Operator Registry
 
-### Implement : gradient operator registry
+A backward network is built up with several backward operators. Backward operators take forward operators' inputs, outputs and output gradients and then calculate its input gradients.
 
-|                        | forward operator | backward operator                |
-| ---------------------- | ---------------- | -------------------------------- |
-| **Operator::inputs_**  | Inputs           | Inputs, Outputs, OutputGradients |
-| **Operator::outputs_** | Outputs          | InputGradients                   |
+-|                        | forward operator | backward operator
```
Comment (dzhwinter) [inline paddle/framework/backward.md]: format error.  `-` appear in the form left.

File: paddle/framework/backward.md [hunk 2]
```diff
@@ -1,23 +1,53 @@
...
 
-Inputs/Outputs means the input/output of the operator,  InputGradients/OutputGradients is the gradient respect to forward opeartor. Forward operator and Backward operator are isomorphic, save their corresponding needs into member attribute.
+ In most cases, there is a one-to-one correspondence between forward and backward operators. These correspondences are recorded by a global hash map(`OpInfoMap`). To follow the philosophy of minimum core and make operators pluggable, the registry mechanism is introduced.
 
-We use a global hash map record the gradient operators available, follow the philosophy  of minimum core, make operator pluggable unit. Each gradient is an operator and it needs to regist itself. 
+For example, we have got a `add_two_op`, and we can register it's information and corresponding backward operator by the following macro:
 
-grad_op_builder(fengjiayi)
+```cpp
+REGISTER_OP(add_two, AddTwoOp, AddTwoOpMaker, add_two_grad, AddTwoGradOp);
```
Comment (dzhwinter) [inline paddle/framework/backward.md]: replace with another operator, please. Otherwise, the user may be confused with the `Generic Add` in backward network construction.

File: paddle/framework/backward.md [hunk 3]
```diff
@@ -1,23 +1,53 @@
-## Operator/expression 's Backward
+# Operator/expression 's Backward
 
-### Motivation
+## Motivation
 
-In Neural Network, the backpropagation algorithm follows the chain rule, so we need to compound the fundmental gradient operators/expressions together with chain rule . Every forward network need a backward network to construct the full computation lineage, the operator/ expression's Backward feature will generate the backward pass respect to forward pass.
+In Neural Network, the backpropagation algorithm follows the chain rule, so we need to compound the fundmental gradient operators/expressions together with chain rule . Every forward network need a backward network to construct the full computation lineage, the operator/expression's backward pass will be generated respect to forward pass.
```
[the remark this answers is not in the document]
Comment (Canpio) [reply paddle/framework/backward.md]: done

File: paddle/framework/backward.md (hunk 1 above)
Comment (Canpio) [reply paddle/framework/backward.md]: done

File: paddle/framework/backward.md (hunk 2 above)
Comment (Canpio) [reply paddle/framework/backward.md]: done
